text/0202-events-and-logs-api.md

```diff
@@ -0,0 +1,187 @@
+LoggerProvider/Logger are analogous to TracerProvider/Tracer.
+
+![Events and Logs API classes](img/0202-events-and-logs-api.png)
+
+### LoggerProvider
+
+Logger can be accessed with an LoggerProvider.
+
+In implementations of the API, the LoggerProvider is expected to be the stateful object that holds any configuration.
+
+Normally, the LoggerProvider is expected to be accessed from a central place. Thus, the API SHOULD provide a way to set/register and access a global default LoggerProvider.
+
+Notwithstanding any global LoggerProvider, some applications may want to or have to use multiple LoggerProvider instances, e.g. to have different configuration (like LogRecordProcessors) for each (and consequently for the Loggers obtained from them), or because it's easier with dependency injection frameworks. Thus, implementations of LoggerProvider SHOULD allow creating an arbitrary number of instances.
+
+#### LoggerProvider operations
+
+The LoggerProvider MUST provide the following functions:
+
+* Get an Logger
+
+##### Get an Logger
+
+This API MUST accept the following parameters:
+
+- `name` (required): This name SHOULD uniquely identify the [instrumentation scope](../glossary.md#instrumentation-scope), such as the [instrumentation library](../glossary.md#instrumentation-library) (e.g. `io.opentelemetry.contrib.mongodb`), package, module or class name.  If an application or library has built-in OpenTelemetry instrumentation, both [Instrumented library](../glossary.md#instrumented-library) and [Instrumentation library](../glossary.md#instrumentation-library) may refer to the same library. In that scenario, the `name` denotes a module name or component name within that library or application. In case an invalid name (null or empty string) is specified, a working Logger implementation MUST be returned as a fallback rather than returning null or throwing an exception, its `name` property SHOULD be set to an empty string, and a message reporting that the specified value is invalid SHOULD be logged. A library implementing the OpenTelemetry API may also ignore this name and return a default instance for all calls, if it does not support "named" functionality (e.g. an implementation which is not even observability-related). A LoggerProvider could also return a no-op Logger here if application owners configure the SDK to suppress telemetry produced by this library.
+
+- `version` (optional): Specifies the version of the instrumentation scope if the scope has a version (e.g. a library version). Example value: 1.0.0.
+- `schema_url` (optional): Specifies the Schema URL that should be recorded in the emitted telemetry
+- `event_domain` (optional): Specifies the domain for the events created, which should be added in the attribute `event.domain` in the instrumentation scope.
+- `include_trace_context` (optional): Specifies whether the Trace Context should automatically be passed on to the events and logs created by the Logger. This SHOULD be false by default.
```

Comment:
This could be an implementation detail of the SDK. Is there a reason why this is part of the API? (some SDKs may not be able to make these links)

Reply:
What choice will the SDK make, as there are use-cases for both? That is the reason we are leaving it to the user to make the choice using this flag.

Reply:
Is there a reason why the user of the API may not want to include the trace context (if it is available)?

Reply:
Consider a user clicking on a random button that results in an error, while an XHR call is in progress. You do not want to record the error event with the trace context of the Span for the XHR call, as they are not related.

Reply:
They would not be associated - in JS the click handler would be seen as separate code path (the async context propagation handles this), or in sync code like Java it would be on a separate thread.

Reply:
let's take this up later during the spec changes.